src/reflect/value.go

```diff
@@ -858,14 +862,21 @@ type MapIter struct {
 	key Value
 	val Value
 
-	valid bool
+	valid        bool
+	keyInterface bool
 }
 
 func (it *MapIter) Key() Value {
 	if !it.valid {
 		panic("reflect.MapIter.Key called on invalid iterator")
 	}
 
+	if it.keyInterface {
+		intf := *(*interface{})(it.key.value)
+		v := ValueOf(intf)
+		return v
```

Comment:
Does this also work with maps that really do have interfaces as map keys? Like `map[interface{}]int`.
~I think there should be tests to show the fixed behavior.~ I see now there are tests in the last commit, but not for maps that really do have interfaces as keys.